benchmark/BenchmarkMain.cpp

```diff
@@ -218,4 +206,99 @@ int main() {
       addTablesToStringstream);
 
   std::cout << visualization.str() << "\n";
-};
+}
+
+/*
+ * @brief Create a nlohmann::json object with all relevant informations
+ *  about the measurments taken by a BenchmarkRecords.
+ */
+nlohmann::json benchmarksToJson(const BenchmarkRecords& records){
+  // The values for all the categories of benchmarks.
+  const std::vector<BenchmarkRecords::RecordEntry>& singleMeasurements =
+    records.getSingleMeasurements();
+  const std::vector<BenchmarkRecords::RecordGroup>& recordGroups =
+    records.getGroups();
+  const std::vector<BenchmarkRecords::RecordTable>& recordTables =
+    records.getTables();
+
+  // Creating the json object. We actually don't want BenchmarkRecords to
+  // be serilized, because that is the management class for measured
+  // benchmarks. We just want the measured benchmarks.
+  return nlohmann::json{{"singleMeasurements", singleMeasurements},
+    {"recordGroups", recordGroups}, {"recordTables", recordTables}};
+}
+
+void writeJsonToFile(nlohmann::json j, std::string fileName) {
+  // Using this constructor, the file is already opened.
+  std::ofstream myFile(fileName, std::ios::out);
+
+  // Error message, if the file couldn't be opened.
+  if (!myFile.is_open()){
+    std::cerr << "Unable to open file '" << fileName << "'.";
+  }
+
+  // nlohmann already has support for things like this.
+  myFile << j;
+
+  myFile.close();
+}
```

Comment:
Damm, that's quite useful. Done in [4c3151f](https://github.com/ad-freiburg/qlever/pull/860/commits/4c3151ff003b3ad3e658dc1591a3ad0b5d0d4609).

P.S.: It still has the std::ios::out, because I need a default option, if the append argument is false. I'm well aware, that it doesn't do anything.